src/containers/SecretsModal/SecretsModal.js

```diff
@@ -370,68 +371,83 @@ export /* istanbul ignore next */ class SecretsModal extends Component {
-            })}
-            keyValues={annotations}
-            minKeyValues={1}
-            invalidFields={invalidFields}
-            onChange={this.handleAnnotationChange}
-            onRemove={this.handleRemove}
-            onAdd={this.handleAdd}
-          />
-          <div
-            ref={el => {
-              this.annotationsEnd = el;
-            }}
-          />
-        </form>
-      </Modal>
+      <>
+        <Modal
+          open={open}
+          className="modal"
+          data-testid="modal"
+          primaryButtonText="Submit"
+          secondaryButtonText="Close"
+          modalHeading="Create Secret"
+          onSecondarySubmit={handleHideModal}
+          onRequestSubmit={this.handleSubmit}
+          onRequestClose={handleHideModal}
+        >
+          <form>
+            <UniversalFields
+              name={name}
+              selectedNamespace={namespace}
+              accessTo={accessTo}
+              handleChangeTextInput={this.handleChangeTextInput}
+              handleChangeAccessTo={this.handleChangeAccessTo}
+              handleChangeNamespace={this.handleChangeNamespace}
+              invalidFields={invalidFields}
+            />
+            <BasicAuthFields
+              username={username}
+              password={password}
+              serviceAccount={serviceAccount}
+              serviceAccounts={serviceAccounts}
+              namespace={namespace}
+              handleChangeTextInput={this.handleChangeTextInput}
+              handleChangeServiceAccount={this.handleChangeServiceAccount}
+              invalidFields={invalidFields}
+            />
+            <KeyValueList
+              legendText={intl.formatMessage({
+                id: 'dashboard.secretsModal.annotations.legendText',
+                defaultMessage: 'Server URL'
+              })}
+              invalidText={intl.formatMessage({
+                id: 'dashboard.secretsModal.annotations.invalidText',
+                defaultMessage: 'Server URL required.'
+              })}
+              ariaLabelKey={intl.formatMessage({
+                id: 'dashboard.secretsModal.annotations.ariaLabelKey',
+                defaultMessage: 'This is the tag Tekton uses for its resources.'
+              })}
+              ariaLabelValue={intl.formatMessage({
+                id: 'dashboard.secretsModal.annotations.ariaLabelValue',
+                defaultMessage: 'This is the URL for the given Tekton resource.'
+              })}
+              keyValues={annotations}
+              minKeyValues={1}
+              invalidFields={invalidFields}
+              onChange={this.handleAnnotationChange}
+              onRemove={this.handleRemove}
+              onAdd={this.handleAdd}
+            />
+            {errorMessage &&
+              (open && (
+                <InlineNotification
+                  kind="error"
+                  title="Error:"
```

Comment:
need to use `react-intl` for user-facing strings
that includes any headings, labels, tooltips, placeholders, etc.

some more on the Modal above

Reply:
ok, updated this now :)